src/main/java/org/jacp/repositry/QuestionRepositoryCriteria.java

```diff
@@ -0,0 +1,42 @@
+package org.jacp.repositry;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.jacp.entity.CompetitionEntity;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+/**
+ * @author saffchen created on 11.03.2024
+ */
+@Repository
+public class QuestionRepositoryCriteria {
+
+    @PersistenceContext
+    private final EntityManager entityManager;
+
+    public QuestionRepositoryCriteria(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public List<Long> getTasksFromCurrentCompetition(Long competitionId) {
```

Comment:
Please try to use Spring Data JPA methods here, because you don't need Criteria API to filter by one field: 
https://docs.spring.io/spring-data/jpa/reference/jpa/query-methods.html#jpa.query-methods.query-creation

Reply:
Fixed